server/neptune/temporal_worker/server.go

```diff
@@ -0,0 +1,267 @@
+		atlantisURL:      parsedURL,
+		atlantisVersion:  atlantisVersion,
+		ctxLogger:        ctxLogger,
+		scope:            scope,
+		closer:           closer,
+		sslCertFile:      sslCertFile,
+		sslKeyFile:       sslKeyFile,
+		statsNamespace:   statsNamespace,
+		temporalHostPort: temporalHostPort,
+	}
+	return config, nil
+}
+
+func newReporter(cfg valid.Metrics) (tally.StatsReporter, error) {
+	if cfg.Statsd == nil {
+		return nil, nil
+	}
+
+	statsdCfg := cfg.Statsd
+	client, err := statsd.NewClientWithConfig(&statsd.ClientConfig{
+		Address: strings.Join([]string{statsdCfg.Host, statsdCfg.Port}, ":"),
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "initializing statsd client")
+	}
+
+	return tallystatsd.NewReporter(client, tallystatsd.Options{}), nil
+}
+
+func (c *Config) buildTemporalClient() (client.Client, error) {
+	certs, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, err
+	}
+	connectionOptions := client.ConnectionOptions{
+		TLS: &tls.Config{
+			RootCAs:    certs,
+			MinVersion: tls.VersionTLS12,
+		},
+	}
+	clientOptions := client.Options{
+		Namespace:         AtlantisNamespace,
+		ConnectionOptions: connectionOptions,
+		MetricsHandler:    temporal_tally.NewMetricsHandler(c.scope),
+	}
+	if c.temporalHostPort != "" {
+		clientOptions.HostPort = c.temporalHostPort
+	}
+	return client.Dial(clientOptions)
+}
+
+func (c *Config) Index(w http.ResponseWriter, _ *http.Request) {
+	err := templates.TemporalWorkerIndexTemplate.Execute(w, templates.IndexData{
+		AtlantisVersion: c.atlantisVersion,
+		CleanedBasePath: c.atlantisURL.Path,
+	})
+	if err != nil {
+		c.ctxLogger.Error(err.Error())
+	}
+}
+
+type Server struct {
+	Logger         logging.Logger
+	SSLCertFile    string
+	SSLKeyFile     string
+	Router         *mux.Router
+	LyftMode       server.Mode
+	Port           int
+	StatsCloser    io.Closer
+	TemporalClient client.Client
+	HealthStatus   int32
+}
+
+// TODO: as more behavior is added into the TemporalWorker package, inject corresponding dependencies
+func NewServer(config *Config) (*Server, error) {
+	temporalClient, err := config.buildTemporalClient()
+	if err != nil {
+		return nil, err
+	}
+	router := mux.NewRouter()
+	router.HandleFunc("/", config.Index).Methods("GET").MatcherFunc(func(r *http.Request, rm *mux.RouteMatch) bool {
+		return r.URL.Path == "/" || r.URL.Path == "/index.html"
+	})
+	if err != nil {
+		return nil, err
+	}
+	server := Server{
+		TemporalClient: temporalClient,
+		Logger:         config.ctxLogger,
+		SSLCertFile:    config.sslCertFile,
+		SSLKeyFile:     config.sslKeyFile,
+		StatsCloser:    config.closer,
+		Router:         router,
+		HealthStatus:   int32(HEALTHY),
+	}
+	return &server, nil
+}
+
+func (s Server) Start() error {
+	s.Router.HandleFunc("/healthz", s.Healthz).Methods("GET")
+	n := negroni.New(&negroni.Recovery{
+		Logger:     log.New(os.Stdout, "", log.LstdFlags),
+		PrintStack: false,
+		StackAll:   false,
+		StackSize:  1024 * 8,
+	})
```

Comment:
let's move this into `NewServer` and just pass in the http Server if we can.

Also, personally I'd probably wrap the http.Server in a proxy struct to handle TLS, to make this cleaner:

```
type httpServerProxy struct {
    SSLCertFile string
    SSLKeyFile string
    http.Server
}

func (p *httpServerProxy) ListenAndServe() error {
    if p.SSLCertFile != "" && p.SSLKeyFile != "" {
      p.Server.ListenAndServeTLS(p.SSLCertFile, p.SSLKeyFile)
    else {
      p.Server.ListenAndServe()
    }
}
```

Could also add a custom constructor for that proxy that handles the middleware and route registration as well